lib/govuk_design_system_formbuilder/elements/submit.rb

```diff
@@ -48,13 +50,18 @@ def padding_class(content_present)
         %(#{brand}-!-margin-right-1) if content_present
       end
 
+      def disabled_class
+        %(#{brand}-button--disabled) if @disabled
+      end
+
       def extra_args
+        disabled_hash = @disabled ? { disabled: 'disabled' } : {}
         {
           formnovalidate: !@validate,
           data: {
             module: %(#{brand}-button), 'prevent-double-click' => @prevent_double_click
           }.select { |_k, v| v.present? }
-        }
+        }.merge!(disabled_hash)
```

Comment:
Can this be simplified slightly? I _think_ this will work in the same way:

```suggestion
        {
          formnovalidate: !@validate,
          disabled: @disabled,
          data: {
            module: %(#{brand}-button), 'prevent-double-click' => @prevent_double_click
          }.select { |_k, v| v.present? }
        }
```

Reply:
My understanding is that disabled attribute with **any** value disables the input, which is why I did it this way. https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute

Reply:
Yeah, you're absolutely right with regards to the HTML, but Rails' tag helper [will discard](https://github.com/rails/rails/blob/master/actionview/lib/action_view/helpers/tag_helper.rb#L77) any [boolean attributes](https://github.com/rails/rails/blob/master/actionview/lib/action_view/helpers/tag_helper.rb#L16) that aren't truthy.

I've done a quick test with the above modification and it appears to work work in the same manner.

The one _danger_ is that someone might try to disable it with `disabled: 'false'` which wouldn't have the desired effect, but that would be in line with Rails' default behaviour (I tested that too).